File: Atividade-em-Sala-21-10-2024/ControleDeAudio.java
public class ControleDeAudio {
    private Integer volume;

    // Construtor da classe
    public ControleDeAudio() {
        this.volume = 2;
    }

    // Lê o volume do objeto
    public Integer lerVolume() {
        return this.volume;
    }

    // Aumenta o volume em 1 caso ele seja menor do que 10
    public void aumentarVolume(){
        if (this.volume < 10){
            this.volume++;
            System.out.println("Volume aumentado...");
            return;
        }

        System.err.println("Erro ao aumentar volume: Volume já está no máximo!");
    }

    // Diminui o volume em 1, mas nunca deixa ele menor que 0
    public void diminuirVolume(){
        if (this.volume > 0){
            this.volume--;
            System.out.println("Volume diminuído...");
            return;
        }

        System.err.println("Erro ao diminuir volume: Volume já está no mínimo!");
    }
    
}

File: Atividade-em-Sala-21-10-2024/controleDeAudio.js
var ControleDeAudio = /** @class */ (function () {
    function ControleDeAudio() {
        this.volume = 2;
    }
    ControleDeAudio.prototype.aumentarVolume = function () {
        if (this.volume < 10) {
            this.volume++;
            console.log("Volume aumentado...");
            return;
        }
        console.log("O volume ja está no máximo!");
    };
    ControleDeAudio.prototype.diminuirVolume = function () {
        if (this.volume > 0) {
            this.volume--;
            console.log("Volume diminuído...");
            return;
        }
        console.log("O volume ja está no mínimo.");
    };
    ControleDeAudio.prototype.lerVolume = function () {
        return this.volume;
    };
    return ControleDeAudio;
}());
var vol = new ControleDeAudio;
console.log("Volume inicial:", vol.lerVolume());
for (var i = 0; i < 9; i++) {
    vol.aumentarVolume();
}
console.log("Volume atualizado: ", vol.lerVolume());
vol.diminuirVolume();
vol.diminuirVolume();
vol.diminuirVolume();
console.log("Volume após diminuição: ", vol.lerVolume());
